devito/types/basic.py

```diff
@@ -1019,29 +1038,33 @@ def _evaluate(self, **kwargs):
         This allow to evaluate off grid points as EvalDerivative that are better
         for the compiler.
         """
+        mapper = self._grid_map
+        subs = mapper.pop('subs', {})
```

Comment:
since `_grid_map` is (must be) a `cached_property`, use `.get()` instead of `.pop()`

In fact, `_grid_map`, now that you make me think twice about it, should return a `frozendict` to be pedantic